aii-pxelinux/src/main/perl/pxelinux.pm

```diff
@@ -67,23 +78,40 @@ sub get_anaconda_version
     return $version;
 }
 
-# Returns the absolute path where the PXE file must be written.
+# Prints an IP address in hexadecimal.
+sub hexip
+{
+    my ($ip, $variant) = @_;
+
+    my $hexip_str='';
+    if ( $variant eq PXE_VARIANT_PXELINUX ) {
+        $hexip_str = sprintf ("%02X%02X%02X%02X", $1, $2, $3, $4) if ($ip =~ m/^(\d+)\.(\d+)\.(\d+)\.(\d+)$/);
```

Comment:
if the `sprintf` statement (without the `grub`part below)  and `if` conditon are the same, start with that. this is again very similar code that is hard to review.